### PR TITLE
Presentation entity, service and migration

### PR DESCRIPTION
**Pull request by VL4DYSL4V**

### CODE REVIEW

**Comment by VL4DYSL4V** [comment]
Look at "Added link..." and the first "Merge remote...", I have no idea why there are three of them

**Review by VolodymyrYavorskyi — APPROVED** [review]

File: slidez-core/build.gradle [hunk 1]
```diff
@@ -32,16 +32,19 @@ dependencies {
...
 	implementation 'org.springframework.boot:spring-boot-starter-actuator'
 	implementation 'org.springframework.boot:spring-boot-starter-web'
-	implementation 'org.springframework.boot:spring-boot-starter-websocket'
+    implementation 'org.springframework.boot:spring-boot-starter-websocket'
 	implementation 'org.springframework.boot:spring-boot-starter-security'
+    implementation 'org.springframework.boot:spring-boot-starter-oauth2-client'
+    // https://mvnrepository.com/artifact/com.google.api-client/google-api-client
+    implementation group: 'com.google.api-client', name: 'google-api-client', version: '1.32.1'
+    //RD stuff
 	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (Anatolii-Ivanov) [inline slidez-core/build.gradle:41]: Why your tabs are not the same as they was)

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/SlidezApplication.java [hunk 2]
```diff
@@ -4,8 +4,11 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
-@SuppressWarnings({"FinalClassCheck", "HideUtilityClassConstructor"}) class SlidezApplication {
+@SuppressWarnings({ "FinalClassCheck", "HideUtilityClassConstructor" })
+class SlidezApplication {
+
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/SlidezApplication.java:9]: Don't change main class

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java [hunk 3]
```diff
@@ -51,7 +52,15 @@
...
-		return new ResponseEntity<>(authResponse.get(), HttpStatus.OK);
+		return new ResponseEntity(authResponse.get(), HttpStatus.OK);
+	}
+
+	@PostMapping("login-by-token")
+	public ResponseEntity<AuthResponse> loginByToken(
+			@RequestBody AuthorizationByTokenRequest authorizationByTokenRequest) {
+		Optional<AuthResponse> authResponseOptional = this.authService.performLoginByToken(authorizationByTokenRequest);
+		return authResponseOptional.map(authResponse -> new ResponseEntity<>(authResponse, HttpStatus.OK))
+				.orElseGet(() -> new ResponseEntity<>(HttpStatus.UNAUTHORIZED));
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java:63]: Why I see here google auth??

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/jwt/JwtProvider.java [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import java.util.Date;
 
 import com.binarystudio.academy.slidez.domain.user.model.User;
+import com.binarystudio.academy.slidez.infrastructure.security.exception.JwtException;
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/jwt/JwtProvider.java:9]: ? You don't need it here

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java (hunk 3 above)
Comment (VladislavArzamastsev) [reply slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java:63]: It only looks like google auth, but it uses different service

Comment (Anatolii-Ivanov) [reply slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java:63]: Oauth* 
You can see in commits that it is here

Comment (VL4DYSL4V) [reply slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java:63]: I got that just now, sorry

**Pull request closed without merging**